src/userEvent/touchEvents.ts

```diff
@@ -0,0 +1,53 @@
+import { ReactTestInstance } from 'react-test-renderer';
+import act from '../act';
+import { getHostParent } from '../helpers/component-tree';
+import { isHostElementPointerEventEnabled } from '../helpers/isHostElementPointerEventEnabled';
+
+const defaultPressEvent = {
+  persist: jest.fn(),
+  nativeEvent: {
+    timestamp: new Date().getTime(),
+  },
+  currentTarget: { measure: jest.fn() },
+};
+
+export type PressOptions = {
+  pressDuration: number;
+};
+
+export const press = (
```

Comment:
It seems that most of the original User Event functions return `Promise<void>`, not sure why though, looks like it has `delay` options that puts a pause between subsequent events to simulator user behaviour. I wonder if that would make sense for our implementation as well.

Reply:
Not sure but I think it's async because it also works with real timers. At least that's the only way I think it could work. Here we only do a press so I don't see where we could add some delay, or maybe at the end? Having some delay while typing text would make sense probably. But then again the more we add delay simulation, the more this would be slow without fake timers so I really feel like we should advocate the usage of fake timers

Reply:
RTL's User Event seems to use `wait` utility to handle both real and fake timers wait:
https://github.com/testing-library/user-event/blob/main/src/utils/misc/wait.ts

```
export function wait(config: Instance['config']) {
  const delay = config.delay
  if (typeof delay !== 'number') {
    return
  }
  return Promise.all([
    new Promise<void>(resolve => globalThis.setTimeout(() => resolve(), delay)),
    config.advanceTimers(delay),
  ])
}
```
Looks like `Promise` + `setTimeout` is used for real timers.

Reply:
Ok it's what I had in mind as well, I'll try to make that work. The only major downside is that the API will be async even for fake timers users but I guess it's not a major issue either

Reply:
RTL's User Event also seems to be async and throws `awaits` in [every](https://testing-library.com/docs/user-event/intro/) [example](https://testing-library.com/docs/user-event/utility). So if we are modelling APIs after them, then let it be so.